### PR TITLE
Added ability to generate image from png file

### PR DESCRIPTION
**Pull request by ScreamingHawk**
User can provide a png file which the program then scales, converts to greyscale, then uses rbg to determine pixel density. 
Also outputs the grey scaled thumbnail. 

Requires PIL or Pillow or equiv

### CODE REVIEW

**Comment by gelstudios** [comment]
This is really cool, is there a graceful way to handle importing PIL? I don't want to make that a hard requirement for casual users if I can help it.

**Comment by ScreamingHawk** [comment]
Yup, you can install Pillow with one command as shown here https://pillow.readthedocs.org/installation.html#basic-installation

**Comment by ScreamingHawk** [comment]
Hey. Still looking to pull this?

File: gitfiti.py [hunk 1]
```diff
@@ -168,6 +170,41 @@ def load_images(img_names):
         loaded_imgs[name] = json.loads(img_list)
         return loaded_imgs
 
+
+max_size = 40, 7
+
+def from_image(fin):
```
Comment (evestera) [inline gitfiti.py]: To make PIL a soft requirement, move the import inside the function where you need it(i.e. here), and do something like this:

    try:
        import PIL
    except ImportError:
        print 'Error: No imaging library is installed'
        print 'Run "pip install Pillow" before using this functionality'
        sys.exit(1)

**Comment by gelstudios** [comment]
Yes absolutely, but in the interest of keeping gitfiti easy to use for everyone, I would strongly prefer to wrap the pil/pillow dependencies in a conditional. That way, the associated import statements are only evaluated if a user actually chooses the "import from image file name" option.

**Comment by homeworkprod** [comment]
How about building a separate transformer tool that parses an image and spits out template data that gitfiti can read?

**Comment by ScreamingHawk** [comment]
Hey, I've updated PIL to be a soft requirement as per evestera's suggestion. Should be good to pull

**Pull request closed without merging**